### ISSUE TITLE
Tutorial 702 WindingNumber

### ISSUE THREAD

**Issue by BruegelN** [libigl/libigl#1222]
#### Describe your issue
Using gcc 8.3.0 (Ubuntu 19.04 default) and gcc  7.4.0 (Ubuntu 18.04 LTS default) I can compile latest master/dev branch (version 2.1.0).
But when I try to run tutorial 702_WindingNumber_bin I get the following error:
```
702_WindingNumber_bin: /home/nico/git/libigl/include/igl/slice.cpp:196: void igl::slice(const Eigen::DenseBase<Derived>&, const Eigen::DenseBase<OtherDerived>&, const Eigen::DenseBase<DerivedC>&, Eigen::PlainObjectBase<DerivedY>&) [with DerivedX = Eigen::Matrix<double, -1, 1>; DerivedR = Eigen::Matrix<int, -1, 1>; DerivedC = Eigen::Matrix<int, -1, 1>; DerivedY = Eigen::Matrix<double, -1, 1>]: Assertion `R.minCoeff() >= 0' failed.
```
And it's the same for the previous release(version 2.0.0).
Using clang on macOS everything works fine.

I only found this bcs. I wanted to move my fork from Eigen 3.2.10 to Eigen 3.3.7 by default and made some tests before and after(same error with both eigen versions).
Maybe will have a closer look at it next week but as of right know it's not my top priority.

#### Check all that apply (change to `[x]`)
- [ ] Windows
- [ ] Mac OS X
- [x] Linux
- [x] I have tried the `dev` branch and the problem persists

See https://libigl.github.io/CONTRIBUTING/#bugreport for more tips.

**Comment by alecjacobson** [comment]
Can you see if it's reading in the files for that tutorial correctly?

>

**Comment by BruegelN** [comment]
@alecjacobson yes it does
```
nico:~/git/libigl/build$ ./tutorial/702_WindingNumber_bin 
Usage:
[space]  toggle showing input mesh, output mesh or slice 
         through tet-mesh of convex hull.
'.'/','  push back/pull forward slicing plane.

Computing winding number over all 164916 tets...
```
then it starts computing (on all 8 threads for quite a while) before I get the error message from above.

**Comment by BruegelN** [comment]
This is the stack trace when I try to run tutorial 702 on my ubuntu machine.
```
__GI_raise 0x00007f326de6ee97
__GI_abort 0x00007f326de70801
__assert_fail_base 0x00007f326de6039a
__GI___assert_fail 0x00007f326de60412
igl::slice<Eigen::Matrix<double, -1, 1, 0, -1, 1>, Eigen::Matrix<int, -1, 1, 0, -1, 1>, Eigen::Matrix<int, -1, 1, 0, -1, 1>, Eigen::Matrix<double, -1, 1, 0, -1, 1> > slice.cpp:196
igl::slice<Eigen::Matrix<double, -1, 1, 0, -1, 1>, Eigen::Matrix<double, -1, 1, 0, -1, 1> > slice.cpp:224
update_visualization main.cpp:46
main main.cpp:146
__libc_start_main 0x00007f326de51b97
_start 0x0000564eccab322a
```

When I replace `big-sigcat.mesh` with `hand.mesh`.
This is a much smaller 3d model, thus faster to compute.
The viewer pops up as expected, I'm able to move the plane with `,`/`.` but if I move for I while it crashes again with the following stack trace on my ubuntu machine.
```
__GI_raise 0x00007f57ca1a5e97
__GI_abort 0x00007f57ca1a7801
__assert_fail_base 0x00007f57ca19739a
__GI___assert_fail 0x00007f57ca197412
igl::slice<Eigen::Matrix<double, -1, 1, 0, -1, 1>, Eigen::Matrix<int, -1, 1, 0, -1, 1>, Eigen::Matrix<int, -1, 1, 0, -1, 1>, Eigen::Matrix<double, -1, 1, 0, -1, 1> > slice.cpp:196
igl::slice<Eigen::Matrix<double, -1, 1, 0, -1, 1>, Eigen::Matrix<double, -1, 1, 0, -1, 1> > slice.cpp:224
update_visualization main.cpp:46
key_down main.cpp:97
std::_Function_handler<bool (igl::opengl::glfw::Viewer&, unsigned int, int), bool (*)(igl::opengl::glfw::Viewer&, unsigned char, int)>::_M_invoke(std::_Any_data const&, igl::opengl::glfw::Viewer&, unsigned int&&, int&&) std_function.h:302
std::function<bool (igl::opengl::glfw::Viewer&, unsigned int, int)>::operator()(igl::opengl::glfw::Viewer&, unsigned int, int) const std_function.h:706
igl::opengl::glfw::Viewer::key_down Viewer.cpp:600
glfw_key_callback Viewer.cpp:87
_glfwInputKey input.c:284
processEvent x11_window.c:1264
_glfwPlatformPollEvents x11_window.c:2698
_glfwPlatformWaitEvents x11_window.c:2724
glfwWaitEvents window.c:1076
igl::opengl::glfw::Viewer::launch_rendering Viewer.cpp:244
igl::opengl::glfw::Viewer::launch Viewer.cpp:131
main main.cpp:148
__libc_start_main 0x00007f57ca188b97
_start 0x0000562562c0022a
```

__However__ I'm now able make I crash on macOS with Clang as well by moving the plane. Both with  `big-sigcat.mesh` and `hand.mesh`. The stack trace like the one above
igl::slice() slice.cpp:196
igl::slice slice.cpp:224
update_visualization main.cpp:46

And indeed `R` always does contain very large negative values. Maybe from integer overflow or double => int conversion.

**Comment by BruegelN** [comment]
Solved by #1337
